### PR TITLE
Fix current compile warnings

### PR DESCRIPTION
**Pull request by xezon**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,5 +1,7 @@
 cmake_minimum_required(VERSION 3.1.0)
 
+set(FETCHCONTENT_QUIET off)
```
Comment (xezon) [inline CMakeLists.txt:3]: This shows some information about the FetchContent feature.

File: CMakeLists.txt [hunk 2]
```diff
@@ -197,7 +199,22 @@ if(USE_GAMEMATH)
...
         FetchContent_Populate(gamemath_git)
+        # disable warning C4146: unary minus operator applied to unsigned type, result still unsigned
+        # disable warning C4244: '=': conversion from 'const double' to 'int32_t', possible loss of data
+        # disable warning C4554: '^': check operator precedence for possible error; use parentheses to clarify precedence
+        # disable warning C4723: potential divide by 0
+        # disable warning C4756: overflow in constant arithmetic
+        if(MSVC)
+            set(gamemath_CMAKE_C_FLAGS ${CMAKE_C_FLAGS})
+            set(CMAKE_C_FLAGS "${gamemath_CMAKE_C_FLAGS} /wd4146 /wd4244 /wd4554 /wd4723 /wd4756")
+        endif()
```
Comment (xezon) [inline CMakeLists.txt]: Changes C flags temporarily to add warning suppression.

File: src/game/logic/system/gamelogic.h [hunk 3]
```diff
@@ -126,7 +126,7 @@ class GameLogic : public SubsystemInterface, public SnapShot
             return nullptr;
         }
 
-        if (id >= m_objectLookupTable.size()) {
+        if (id >= static_cast<ObjectID>(m_objectLookupTable.size())) {
```
Comment (xezon) [inline src/game/logic/system/gamelogic.h]: This comparison caused signed/unsigned mismatch warning.

Comment (OmniBlade) [reply src/game/logic/system/gamelogic.h]: Does this still cause a warning even after the underlying type of the enum was changed if you ommit the cast?

Comment (xezon) [reply src/game/logic/system/gamelogic.h]: on Win x86 the cast would no longer be necessary. On Win x64 it should be fine as well. size_t would be 64 bit unsigned as opposed to 32 bit unsigned then. I suspect static_cast could be omitted here. If there was still a warning on another platform, we could look into it again.

Comment (OmniBlade) [reply src/game/logic/system/gamelogic.h]: Sounds good to me.

File: CMakeLists.txt (hunk 2 above)
Comment (OmniBlade) [reply CMakeLists.txt]: I've updated gamemath to silence these upstream, fetchcontent commit hash just needs updating now instead.

**Comment by OmniBlade** [comment]
Can you rebase your commits to remove the merge commit from the history or just remove it as the merge will rebase the whole PR on top of develop anyhow? I'm trying to keep the history linear on develop.

**Comment by xezon** [comment]
Is it even possible to rebase? The changes are already on GitHub.

**Comment by duncanspumpkin** [comment]
Yes it's always possible to rebase. You just need to force push your changes. 

```
git rebase develop
git push origin <branch> --force
```

**Comment by janisozaur** [comment]
Would be nice to have proper commit messages.

https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
https://chris.beams.io/posts/git-commit/
https://git-scm.com/docs/git-commit

Or pretty much any other result for "git commit messages" in a search engine.

**Comment by xezon** [comment]
Thanks. The force push did the trick.

**Review by OmniBlade — APPROVED** [review]

**Pull request merged**